sdk/storage/azure-storage-blob/tests/test_get_blob.py

```diff
@@ -144,6 +146,50 @@ def test_get_blob_to_bytes(self, resource_group, location, storage_account, stor
         # Assert
         self.assertEqual(self.byte_data, content)
 
+    @GlobalStorageAccountPreparer()
+    def test_get_blob_in_raw_stream(self, resource_group, location, storage_account, storage_account_key):
+        # parallel tests introduce random order of requests, can only run live
+        self._setup(storage_account, storage_account_key)
+        blob_name = self._get_blob_reference()
+        blob = self.bsc.get_blob_client(self.container_name, blob_name)
+        file_path = "C:/Users/xiafu/Downloads/testgzip.txt.gz"
+        data = self.get_random_bytes(1024*1024*1024)
+
+        with gzip.open(file_path, 'wb') as stream2:
+            stream2.write(data)
+
+        with open(file_path, 'rb') as stream:
+            data = stream.read()
+            blob.upload_blob(data, content_settings=ContentSettings(content_encoding='gzip'), overwrite=True)
+
+        content = blob.download_blob(offset=1024, length=512, decompress=False).readall()
+
+        # Assert
+        self.assertEqual(data[1024: 1024 + 512], content)
+
+    @pytest.mark.live_test_only
+    @GlobalStorageAccountPreparer()
+    def test_get_blob_in_raw_stream(self, resource_group, location, storage_account, storage_account_key):
```

Comment:
this would fail once we turn on recording because content_encoding header is set and recording mechanism always tries to decompress.